backend/oasst_backend/api/v1/messages.py

```diff
@@ -16,38 +16,90 @@
 
 @router.get("/", response_model=list[protocol.Message])
 def query_messages(
-    username: str = None,
-    api_client_id: str = None,
-    max_count: int = Query(10, gt=0, le=1000),
-    start_date: datetime.datetime = None,
-    end_date: datetime.datetime = None,
-    only_roots: bool = False,
-    desc: bool = True,
-    allow_deleted: bool = False,
+    auth_method: Optional[str] = None,
+    username: Optional[str] = None,
+    api_client_id: Optional[str] = None,
+    max_count: Optional[int] = Query(10, gt=0, le=1000),
+    start_date: Optional[datetime.datetime] = None,
+    end_date: Optional[datetime.datetime] = None,
+    only_roots: Optional[bool] = False,
+    desc: Optional[bool] = True,
+    allow_deleted: Optional[bool] = False,
     api_client: ApiClient = Depends(deps.get_api_client),
     db: Session = Depends(deps.get_db),
 ):
     """
     Query messages.
     """
-    start_date = unaware_to_utc(start_date)
-    end_date = unaware_to_utc(end_date)
-
     pr = PromptRepository(db, api_client)
-    messages = pr.query_messages(
+    messages = pr.query_messages_ordered_by_created_date(
+        auth_method=auth_method,
         username=username,
         api_client_id=api_client_id,
         desc=desc,
         limit=max_count,
-        start_date=start_date,
-        end_date=end_date,
+        gte_created_date=start_date,
+        lte_created_date=end_date,
         only_roots=only_roots,
         deleted=None if allow_deleted else False,
     )
 
     return utils.prepare_message_list(messages)
 
 
+@router.get("/cursor", response_model=protocol.MessagePage)
+def get_messages_cursor(
+    lt: Optional[str] = None,
+    gt: Optional[str] = None,
+    user_id: Optional[UUID] = None,
+    auth_method: Optional[str] = None,
+    username: Optional[str] = None,
+    api_client_id: Optional[str] = None,
+    only_roots: Optional[bool] = False,
+    include_deleted: Optional[bool] = False,
+    max_count: Optional[int] = Query(10, gt=0, le=1000),
+    desc: Optional[bool] = False,
+    api_client: ApiClient = Depends(deps.get_api_client),
+    db: Session = Depends(deps.get_db),
+):
+    def split_cursor(x: str | None) -> tuple[datetime.datetime, UUID]:
+        if not x:
+            return None, None
+        f = x.split("$", maxsplit=1)
+        if len(f) == 2 and len(f[0]) == 36:
```

Comment:
other than `36`, maybe `len("some_constant")` or a comment why 36

Reply:
ok, 36 is the length of a full UUID converted with str() (with dashes) ... I could also validate with a regex ..